core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -166,17 +184,28 @@ public Schema array(Schema array, Supplier<Schema> element) {
         Schema keyValueSchema = array.getElementType();
         Schema.Field keyField = keyValueSchema.getFields().get(0);
         Schema.Field valueField = keyValueSchema.getFields().get(1);
-        Schema.Field valueProjection = element.get().getField("value");
+        Schema elementSchema = element.get();
+
+        if (fieldId(valueField) == null) {
```

Comment:
What if the key field has no ID? In that case, I think it should also not be possible to project the field.